node/src/service.rs

```diff
@@ -827,7 +875,103 @@ pub async fn start_node_impl_container(
 
     start_network.start_network();
 
-    Ok((task_manager, client))
+    Ok((task_manager, client, start_collation))
+}
+
+// Copy of `cumulus_client_service::start_collator`, that doesn't fully start the collator: it is
+// missing the final call to `cumulus_client_collator::start_collator`.
+// Returns the params of the call to `cumulus_client_collator::start_collator`.
+pub fn partial_start_collator<'a, Block, BS, Client, Backend, RCInterface, Spawner>(
```

Comment:
Hopefully we can in the future propose a pr to cumulus to make it more modular

Reply:
But maybe we can move it to a different file in a follow-up refactor

Reply:
I think this is too specific to be accepted upstream, however we should try to add `Clone` to `StartCollatorParams`, it will save us some lines of code.